### PR TITLE
Add connection details to azure_rm_privateendpoint_info

### PR DESCRIPTION
**Pull request by dhageman**
##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
This code modifies azure_rm_privateendpoint_info to return the connection details associated with a private endpoint.  This is necessary for private endpoints that don't connect to a private link service as the newly added azure_rm_privateendpointconnection_info only supports those type of links.  The primary motivation for this feature is to allow identification of private endpoints that are pointing to resources which no longer exist. 

##### ADDITIONAL INFORMATION
This pull request introduces a change in the output of the command.  It will return a complex type instead of an array of connection id strings.

Old result:
```
                "private_link_service_connections": [
                    "/subscriptions/xxx-xxx-xxx/resourceGroups/rg-infrastructure/providers/Microsoft.Network/privateEndpoints/pe-aks-z6fjatiyrb3kc-6addcd51/privateLinkServiceConnections/pe-aks-z6fjatiyrb3kc-6addcd51"
                ],

```
New result:
```
                "private_link_service_connections": [
                    {
                        "connection_state": {
                            "actions_required": "None",
                            "description": "Auto Approved",
                            "status": "Disconnected"
                        },
                        "group_ids": [
                            "management"
                        ],
                        "id": "/subscriptions/xxx-xxx-xxxx/resourceGroups/rg-infrastructure/providers/Microsoft.Network/privateEndpoints/pe-aks-z6fjatiyrb3kc--6addcd51/privateLinkServiceConnections/pe-aks-z6fjatiyrb3kc-6addcd51",
                        "name": "pe-aks-z6fjatiyrb3kc-6addcd51",
                        "type": "Microsoft.Network/privateEndpoints/privateLinkServiceConnections"
                    }
                ],
```

### CODE REVIEW

File: plugins/modules/azure_rm_privateendpoint_info.py [hunk 1]
```diff
@@ -116,8 +116,49 @@
             description:
                 - The resource id of the private endpoint to connect.
             returned: always
+            type: complex
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_privateendpoint_info.py:119]: duplicate
```suggestion

```

File: plugins/modules/azure_rm_privateendpoint_info.py [hunk 2]
```diff
@@ -116,8 +116,49 @@
...
+                            returned: always
+                            type: str
+                            sample: Approved
+                        actions_required:
+                            description:
+                                - A message indicating if changes on the service provider require any updates on the consumer.
+                            type: str
+                            returned: always
+                            sample: "This is action_required string"
+        group_ids:
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_privateendpoint_info.py]: Then move eight Spaces。
```suggestion
                group_ids:
                    description:
                        - List of group_ids associated with private endpoint
                    returned: always
                    type: list
                    sample: [""postgresqlServer"]
``

File: plugins/modules/azure_rm_privateendpoint_info.py [hunk 3]
```diff
@@ -116,8 +116,54 @@
...
+                    returned: always
+                    type: list
+                    sample: [""postgresqlServer"]
+``
+            description:
+                - List of group_ids associated with private endpoint
+            returned: always
             type: list
-            sample: ["/subscriptions/xxx/resourceGroups/myRG/providers/Microsoft.Network/privateEndpoints/point/privateLinkServiceConnections/point",]
+            sample: "management"
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_privateendpoint_info.py]: type is list!
```suggestion
            sample: ["management"]
```

File: plugins/modules/azure_rm_privateendpoint_info.py [hunk 4]
```diff
@@ -116,8 +116,54 @@
             description:
                 - The resource id of the private endpoint to connect.
             returned: always
+            returned: always
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_privateendpoint_info.py]:
```suggestion
            type: complex
```

File: plugins/modules/azure_rm_privateendpoint_info.py [hunk 5]
```diff
@@ -116,8 +116,54 @@
...
+                                - A message indicating if changes on the service provider require any updates on the consumer.
+                            type: str
+                            returned: always
+                            sample: "This is action_required string"
+                group_ids:
+                    description:
+                        - List of group_ids associated with private endpoint
+                    returned: always
+                    type: list
+                    sample: [""postgresqlServer"]
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_privateendpoint_info.py]:
```suggestion
                    sample: ["postgresqlServer"]
```

File: plugins/modules/azure_rm_privateendpoint_info.py [hunk 6]
```diff
@@ -116,8 +116,54 @@
...
+``
+            description:
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_privateendpoint_info.py]:
```suggestion

```

File: plugins/modules/azure_rm_privateendpoint_info.py [hunk 7]
```diff
@@ -116,8 +116,54 @@
...
+                - List of group_ids associated with private endpoint
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_privateendpoint_info.py]:
```suggestion

```

File: plugins/modules/azure_rm_privateendpoint_info.py [hunk 8]
```diff
@@ -116,8 +116,54 @@
...
+            returned: always
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_privateendpoint_info.py]:
```suggestion

```

File: plugins/modules/azure_rm_privateendpoint_info.py [hunk 9]
```diff
@@ -116,8 +116,54 @@
...
             type: list
-            sample: ["/subscriptions/xxx/resourceGroups/myRG/providers/Microsoft.Network/privateEndpoints/point/privateLinkServiceConnections/point",]
+            sample: ["management"]
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_privateendpoint_info.py]:
```suggestion

```

**Review by Fred-sun — COMMENTED** [review]
Delete lines 160 to 166

File: plugins/modules/azure_rm_privateendpoint_info.py [hunk 10]
```diff
@@ -116,8 +116,50 @@
...
+                            returned: always
+                            sample: "This is action_required string"
+                group_ids:
+                    description:
+                        - List of group_ids associated with private endpoint
+                    returned: always
+                    type: list
+                    sample: ["postgresqlServer"]
+``
             type: list
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_privateendpoint_info.py]: delete 162 line!

**Review by xuzhang3 — APPROVED** [review]

**Comment by xuzhang3** [comment]
@dhageman LGTM 🚢

**Pull request merged**